artifactory/commands/golang/go.go

```diff
@@ -122,6 +127,29 @@ func (gc *GoCommand) Run() error {
 		return err
 	}
 	if isCollectBuildInfo {
+		tempDirPath := ""
+		if isGoGetCommand := len(gc.goArg) > 0 && gc.goArg[0] == "get"; isGoGetCommand {
+			if len(gc.goArg) < 2 {
+				// Package name was not supply, invalid go get commend
+				return errorutils.CheckError(errors.New("invalid get command, package name is missing"))
+			}
+			rtDetails, err := resolverDetails.CreateArtAuthConfig()
+			tempDirPath, err = fileutils.CreateTempDir()
+			if err != nil {
+				return err
+			}
+			err = copyGoPackageFiles(tempDirPath, gc.goArg[1], gc.resolverParams.TargetRepo(), rtDetails)
+			if err != nil {
+				return err
+			}
+			// Cleanup the temp working directory
+			defer fileutils.RemoveTempDir(tempDirPath)
```

Comment:
This defer should be moved to line 141 (right after the ```CreateTempDir``` error check), to make sure the temp dir is deleted in every scenario.